wallet/src/account/output_cache/mod.rs

```diff
@@ -151,4 +187,84 @@ impl OutputCache {
 
         utxos
     }
+
+    pub fn get_abandonable_transactions(&self) -> Vec<&WithId<Transaction>> {
+        self.txs
+            .values()
+            .filter_map(|tx| match tx {
+                WalletTx::Block(_) => None,
+                WalletTx::Tx(tx) => match tx.state() {
+                    TxState::Inactive => Some(tx.get_transaction_with_id()),
+                    TxState::Confirmed(_, _)
+                    | TxState::Conflicted(_)
+                    | TxState::InMempool
+                    | TxState::Abandoned => None,
+                },
+            })
+            .collect()
+    }
+
+    pub fn abandon_transaction(&mut self, tx_id: Id<Transaction>) -> WalletResult<()> {
+        let mut to_abandon = BTreeSet::new();
+        to_abandon.insert(OutPointSourceId::from(tx_id));
+
+        while let Some(outpoint_source_id) = to_abandon.pop_first() {
+            if let Some(descendants) = self.unconfirmed_descendants.remove(&outpoint_source_id) {
```

Comment:
Do we have tests for abandoning transactions with descendants?

Reply:
yes